src/main/java/de/tum/in/www1/artemis/service/scheduled/AthenaScheduleService.java

```diff
@@ -0,0 +1,111 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.enumeration.ExerciseLifecycle;
+import de.tum.in.www1.artemis.repository.TextExerciseRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.ExerciseLifecycleService;
+import de.tum.in.www1.artemis.service.connectors.athena.AthenaSubmissionSendingService;
+import tech.jhipster.config.JHipsterConstants;
+
+@Service
+@Profile("athena & scheduling")
+public class AthenaScheduleService {
+
+    private final Logger log = LoggerFactory.getLogger(AthenaScheduleService.class);
+
+    private final ExerciseLifecycleService exerciseLifecycleService;
+
+    private final TextExerciseRepository textExerciseRepository;
+
+    private final Environment env;
+
+    private final Map<Long, ScheduledFuture<?>> scheduledAthenaTasks = new HashMap<>();
+
+    private final AthenaSubmissionSendingService athenaSubmissionSendingService;
+
+    public AthenaScheduleService(ExerciseLifecycleService exerciseLifecycleService, TextExerciseRepository textExerciseRepository, Environment env,
+            AthenaSubmissionSendingService athenaSubmissionSendingService) {
+        this.exerciseLifecycleService = exerciseLifecycleService;
+        this.textExerciseRepository = textExerciseRepository;
+        this.env = env;
+        this.athenaSubmissionSendingService = athenaSubmissionSendingService;
+    }
+
+    @PostConstruct
+    private void scheduleRunningExercisesOnStartup() {
+        final Collection<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
+        if (activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
```

Comment:
just an optional alternative: We already have a ```ProfilService``` that contains a method ```isDev```. In  my opinion that's a bit cleaner.

Reply:
That's much better, nice! I didn't know about the `ProfileService` before.